course/src/isometry.cpp

```diff
@@ -2,16 +2,184 @@
  * Isometry library
  * Author: Agustin Alba Chicar, 2019
  * Author: Gerardo Puga, 2020
+ * Copyright 2020 Ekumen"
  */
 
 #include <isometry/isometry.hpp>
 
 namespace ekumen {
 namespace math {
 
-/*
- * your code here
- */
+Vector3::Vector3(std::initializer_list<double> list) {
+  if (list.size() != 3) {
+    throw std::out_of_range("Incorrect size of list");
+  }
+  auto it = list.begin();
+  x_ = *it;
+  ++it;
+  y_ = *it;
+  ++it;
+  z_ = *it;
+}
+
+double Vector3::operator[](const int i) const {
+  if ((i < 0) || (i > 2)) {
+    throw std::out_of_range("Operator out of range");
+  }
+  if (i == 0) {
+    return x_;
+  } else if (i == 1) {
+    return y_;
+  } else {
+    return z_;
+  }
+}
+
+double& Vector3::operator[](const int i) {
+  if ((i < 0) || (i > 2)) {
+    throw std::out_of_range("Operator out of range");
+  }
+  if (i == 0) {
+    return x_;
+  } else if (i == 1) {
+    return y_;
+  } else {
+    return z_;
+  }
+}
+
+Vector3 Vector3::kUnitX{Vector3(1., 0., 0.)};
+
+Vector3 Vector3::kUnitY{Vector3(0, 1, 0)};
+
+Vector3 Vector3::kUnitZ{Vector3(0, 0, 1)};
+
+Vector3 Vector3::kZero{Vector3(0, 0, 0)};
+
+bool Vector3::operator==(const Vector3& r_vector) const {
+  bool result = false;
+  if (almost_equal(x_, r_vector.x_, 2) && almost_equal(y_, r_vector.y_, 2) &&
+      almost_equal(z_, r_vector.z_, 2)) {
+    result = true;
+  }
+
+  return result;
+}
```

Comment:
you could replace it with just one line.

```c++
bool Vector3::operator==(const Vector3& r_vector) const {
       return almost_equal(x_, r_vector.x_, 2) && almost_equal(y_, r_vector.y_, 2) &&  almost_equal(z_, r_vector.z_, 2);
}
```

Reply:
Making this change activates a compiler message that says that the lines  should be `<=80`.

Reply:
Have you used the clang-format script?

Reply:
The result of using it is that sentence split in two lines.